packages/core/js-client/src/api.ts

```diff
@@ -15,182 +15,158 @@
   def: FunctionCallDef,
   script: string,
 ): Promise<unknown> => {
-  const { args, client: peer, config } = extractFunctionArgs(rawFnArgs, def);
-
-  return callAquaFunction({
-    args,
-    def,
-    script,
-    config,
-    peer,
-  });
-};
+  const argNames = Object.keys(def.arrow);
+  const argCount = argNames.length;
 
-/**
- * Convenience function to support Aqua `service` generation backend
- * The compiler only need to generate a call the function and provide the corresponding definitions and the air script
- * @param args - raw arguments passed by user to the generated function
- * TODO: dont forget to add jsdoc for new arg
- * @param def - service definition generated by the Aqua compiler
- */
-export const v5_registerService = (args: unknown[], def: ServiceDef): void => {
-  // TODO: Support this in aqua-to-js package
-  // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
-  const service: ServiceImpl = args.pop() as ServiceImpl;
+  const functionArgs: Record<string, SimpleTypes | ArrowWithoutCallbacks> =
+    def.arrow.domain.tag === "nil" ? {} : def.arrow.domain.fields;
 
-  const { peer, serviceId } = extractServiceArgs(args, def.defaultServiceId);
+  let peer: FluencePeer | undefined;
 
-  registerService({
-    def,
-    service,
-    serviceId,
-    peer,
-  });
-};
-
-function isConfig(arg: unknown): arg is FnConfig {
-  return typeof arg === "object" && arg !== null;
-}
-
-/**
- * Arguments could be passed in one these configurations:
- * [...actualArgs]
- * [peer, ...actualArgs]
- * [...actualArgs, config]
- * [peer, ...actualArgs, config]
- *
- * This function select the appropriate configuration and returns
- * arguments in a structured way of: { peer, config, args }
- */
-function extractFunctionArgs(
-  args: unknown[],
-  def: FunctionCallDef,
-): {
-  client: FluencePeer;
-  config: FnConfig;
-  args: PassedArgs;
-} {
-  const argumentTypes = getArgumentTypes(def);
-  const argumentNames = Object.keys(argumentTypes);
-  const numberOfExpectedArgs = argumentNames.length;
-
-  let peer: FluencePeer;
-  let config: FnConfig;
-
-  if (isFluencePeer(args[0])) {
+  if (args[0] instanceof FluencePeer) {
     peer = args[0];
     args = args.slice(1);
   } else {
-    if (Fluence.defaultClient == null) {
-      throw new Error(
-        "Could not register Aqua service because the client is not initialized. Did you forget to call Fluence.connect()?",
-      );
-    }
-
     peer = Fluence.defaultClient;
   }
 
-  const maybeConfig = args[numberOfExpectedArgs];
-
-  if (isConfig(maybeConfig)) {
-    config = maybeConfig;
-  } else {
-    config = {};
-  }
-
-  const structuredArgs = args.slice(0, numberOfExpectedArgs);
+  // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
+  const config =
+    argCount < args.length
+      ? // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
+        (args.pop() as CallAquaFunctionConfig | undefined)
```

Comment:
can you please verify that this config adheres to the `CallAquaFunctionConfig` type?

Otherwise it might shoot unexpectedly.